polynote-kernel/src/main/scala/polynote/kernel/PolyKernel.scala

```diff
@@ -111,51 +144,51 @@ class PolyKernel private[kernel] (
                         val newDetail = Option(detail).getOrElse(taskInfo.detail)
                         statusUpdates.publish1(UpdatedTasks(taskInfo.copy(detail = newDetail, progress = (progress * 255).toByte) :: Nil)).unsafeRunSync()
                     }
-                    kernel.runCode(
+
+                    symbolTable.drain() *> interp.runCode(
```

Comment:
Just curious, why did you move `symbolTable.drain()` down here?

Reply:
Because that's really the spot where it needs to happen (i.e. just before the code is run)